EPRN.Portal/Views/Waste/SubTypes.cshtml

```diff
@@ -0,0 +1,102 @@
+                    </div>
+                </div>
+            }
+        }
+
+        @for (int i = 0; i < Model.WasteSubTypeOptions.Count; i++)
+        {
+            <input type="hidden" name="WasteSubTypeOptions[@i].Id" value="@Model.WasteSubTypeOptions[i].Id" />
+            <input type="hidden" name="WasteSubTypeOptions[@i].Name" value="@Model.WasteSubTypeOptions[i].Name" />
+            <input type="hidden" name="WasteSubTypeOptions[@i].Adjustment" value="@Model.WasteSubTypeOptions[i].Adjustment" />
+            <input type="hidden" name="WasteSubTypeOptions[@i].AdjustmentPercentageRequired" value="@Model.WasteSubTypeOptions[i].AdjustmentPercentageRequired.ToString().ToLower()" />
+        }
+    </radios>
+
+    <button class="govuk-button" data-module="govuk-button" type="submit">
+        @MasterResources.ContinueBtn
+    </button>
+</form>
+
+
+@section scripts {
+    <script>
+        $(document).ready(function () {
+            function toggleCustomPercentageVisibility() {
+                var selectedRadio = $('input[type=radio][name="SelectedWasteSubTypeId"]:checked');
+                var customPercentageInput = $('input[type=text][name="CustomPercentage"]');
+                var customPercentageDiv = $('#custom-percentage-div');
+
+                if (selectedRadio.length > 0) {
+                    var showCustomPercentage = selectedRadio.data('custompercentage') === true;
+                    customPercentageDiv.toggle(showCustomPercentage);
+                    var errorSummary = $('.govuk-error-summary');
+                    // Clear the value and validation errors if hidden
+                    if (!showCustomPercentage) {
+                        customPercentageInput.val('');
+                        errorSummary.addClass('govuk-visually-hidden');
+
+                    } else {
+                        if (validationErrorsExist()) {
+                            errorSummary.removeClass('govuk-visually-hidden');
+                        }
+                    }
+                } else {
+                    // If no radio is selected, hide the custom percentage input and clear the value
+                    customPercentageDiv.hide();
+                    customPercentageInput.val('');
+                }
+            }
+
+            function validationErrorsExist() {
+                var errorList = $('.govuk-list.govuk-error-summary__list');
+                var liElements = errorList.find('li');
+
+                return !((liElements.length === 1 && liElements.text().trim() === '') || liElements.length === 0);
+            }
+
+            // Initial visibility setup
+            toggleCustomPercentageVisibility();
+
+            // Event handler for radio button change
+            $('input[type=radio][name="SelectedWasteSubTypeId"]').change(function () {
+                toggleCustomPercentageVisibility();
+            });
+        });
+    </script>
```

Comment:
Put javascript into a javascript file so that it is bundled and minified into a single file